sanic/headers.py

```diff
@@ -30,6 +33,121 @@
 # For more information, consult ../tests/test_requests.py
 
 
+def parse_arg_as_accept(f):
+    def func(self, other):
+        if not isinstance(other, Accept):
+            other = Accept.parse(other)
+        return f(self, other)
+
+    return func
+
+
+class MediaType(str):
+    def __new__(cls, value: str):
+        return str.__new__(cls, value)
+
+    def __init__(self, value: str) -> None:
+        self.value = value
+        self.is_wildcard = self.check_if_wildcard(value)
+
+    def __eq__(self, other):
+        other_is_wildcard = (
+            other.is_wildcard
+            if isinstance(other, MediaType)
+            else self.check_if_wildcard(other)
+        )
+        other_value = other.value if isinstance(other, MediaType) else other
+        return (
+            self.value == other_value or self.is_wildcard or other_is_wildcard
+        )
```

Comment:
Suggest `if ... return True` on wildcard checks to avoid useless processing.

Reply:
As soon as the first condition is met, it will exit. If `self.value == other_value` is `True`, Python will not proceed to `self.is_wildcard` etc.

```python
>>> foo = 123
>>> bar = 123
>>> if foo == bar or foo.something_that_does_not_exist:
...     print("hi")
... 
hi
```